server/functions/src/index.ts

```diff
@@ -12,31 +13,114 @@ const COUNTRY_CODE = /^[A-Z][A-Z]$/; // Literal regex for immediate compilation.
 // https://stackoverflow.com/a/12502351/1509221
 const FCM_TOKEN_MAX_LENGTH = 4096;
 
-enum Platform {
-  IOS = 0,
-  ANDROID = 1,
-  WEB = 2,
-}
-
 interface Client {
   uuid: string;
   token: string;
   disableNotifcations: boolean;
-  platform: Platform;
+  platform: who.Platform;
   isoCountryCode: string;
   subscribedTopics: string[];
 }
 
+refreshcasestats.setDb(db); /// This seems wrong???????????????????????????????????????????????????????????
+export const refreshCaseStats = refreshcasestats.refreshCaseStats;
+
+async function getCaseStatsAsync(req: who.GetCaseStatsRequest) {
+  let jurisdictions = req.jurisdictions;
+  if (jurisdictions == undefined) {
+    throw Error("Bad request.");
+  }
+  let resp = {
+    globalStats: undefined,
+    jurisdictionStats: [],
+    ttl: 0,
+  } as who.GetCaseStatsResponse;
```

Comment:
Consider this slight change:
```
  let resp: who.GetCaseStatsResponse = {
    globalStats: undefined,
    jurisdictionStats: [],
    ttl: 0,
  }
```

... Since the `as` type assertion will _not_ throw a type error if you're missing required fields (I'm basing that on e.g. [this](https://basarat.gitbook.io/typescript/type-system/type-assertion#assertion-considered-harmful)), but when constructing an object of a given type you _will_ be warned about that.

Reply:
thanks! i didn't know how to do 'constructors' but copied another example in this file.